synthesizer/src/vm/finalize.rs

```diff
@@ -852,10 +854,24 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
         // Initialize the list of deployment payers.
         let mut deployment_payers: IndexSet<Address<N>> = Default::default();
 
-        // Abort the transactions that are have duplicates or are invalid. This will prevent the VM from performing
-        // verification on transactions that would have been aborted in `VM::atomic_speculate`.
+        // Abort fee transactions and invalid deployments. This will prevent the
+        // VM from performing expensive verification on invalid fees or
+        // deployments that would have been aborted in `VM::atomic_speculate`.
         for transaction in transactions.iter() {
-            // Determine if the transaction should be aborted.
+            // If the transaction is an Execution, we will fully verify it.
+            // Executions require full verification to avoid malleability attacks.
+            if transaction.is_execute() {
+                executions.push(*transaction);
+                continue;
+            }
```

Comment:
I don't believe we want this. This opens up the attack where malicious parties and induce load on validators by verifying many transactions for free.

To have duplicate `input_id`, `output_ids`, and `tcm` you need to use the same private key to create the transaction, so it would only affect their own transactions.

The malleability attack from validators can only be done via converting the Execution to a Fee transaction. Which your subsequent check already catches.